### PR TITLE
feat: Add ability to disable lifecycle prevent_destroy

### PR DESCRIPTION
**Pull request by grahamhar**
This is a proposal to fix the issue described in #42 

I have used features introduced in Python 3.5 and added a check in case people are using older versions of Python. If you think this needs to work on any version of Python 3 I'm happy to change the approach.

I have also only tested on MacOS, I assume in CI the tests ran under Linux. I'll try to get one of my colleagues who works on Windows to test this.

### CODE REVIEW

File: tftest.py [hunk 1]
```diff
@@ -343,10 +348,23 @@ def setup(self, extra_files=None, plugin_dir=None, init_vars=None,
...
 
     Returns:
       Terraform init output.
     """
+    # remove lifecycle prevent destroy
+    if disable_prevent_destroy:
+      min_python = (3, 5)
+      if sys.version_info < min_python:
+        raise TerraformTestError('The disable_prevent_destroy flag requires at least Python 3.5')
+      for tf_file in Path(self.tfdir).rglob('*.tf'):
```
Comment (ludoo) [inline tftest.py:361]: Do you think it's worth trapping potential file access errors here, and returning a meaningful message?

**Review by ludoo — APPROVED** [review]
Thanks for this! One minor doubt on exception handling when manipulating files, and I'm also wondering if it would make sense to use `pathlib` everywhere for consistency, now that you introduced it. It does not need to happen in this PR, of course, I would just like your opinion.

File: tftest.py (hunk 1 above)
Comment (grahamhar) [reply tftest.py:361]: I did consider wrapping this in try/except for file access type issues, but I thought the script should fail if it couldn't carry out the required actions and the exception raised should be meaningful.

If you think it makes the failure easier to understand I'm happy to add this in, were you thinking something like this:


```python
        try:
          shutil.copy(str(tf_file), f'{str(tf_file)}.bkp')
        except OSError as exc:
          _LOGGER.exception(f'Unable to backup terraform file {tf_file.resolve()}')
          raise TerraformTestError(f'Backup of terraform file ({tf_file.resolve()}) failed')
```

Comment (ludoo) [reply tftest.py:361]: I just prefer seeing a meaningful message, and raising exceptions that use or extend the same base class, makes it easier to read and to consume. But it's not super important, feel free of course to merge as-is, we can revisit later.

Comment (grahamhar) [reply tftest.py:361]: I'd rather do it now, not super hard. Let me update the PR.

Comment (grahamhar) [reply tftest.py:365]: Pushed the updates let e know if that works.

**Review by ludoo — APPROVED** [review]
Thanks a lot for this!

**Comment by grahamhar** [comment]
Any idea when a version for this will be released?

**Comment by ludoo** [comment]
Actually had the tab open to remind me to do it, let me try and do it now.

**Comment by ludoo** [comment]
Done! https://pypi.org/project/tftest/1.6.4/

**Pull request merged**